src/devSVG.cpp

```diff
@@ -124,6 +131,37 @@ inline void write_attrs_linetype(FILE* f, int lty, double lwd, int col) {
     fputs("'", f);
     break;
   }
+
+  // Set line end shape
+  switch(lend)
+  {
+  case GE_ROUND_CAP:
+    write_attr_str(f, "stroke-linecap", "round");
+    break;
+  case GE_SQUARE_CAP:
+    write_attr_str(f, "stroke-linecap", "square");
+    break;
+  case GE_BUTT_CAP: // doing nothing, default is butt
+  default:
+    break;
+  }
+
+  // Set line join shape
+  switch(ljoin)
+  {
+  case GE_ROUND_JOIN:
+    write_attr_str(f, "stroke-linejoin", "round");
+    break;
+  case GE_BEVEL_JOIN:
+    write_attr_str(f, "stroke-linejoin", "bevel");
+    break;
+  case GE_MITRE_JOIN: // We don't need to write "stroke-linejoin" attribute here,
+                      // since the default is "miter". However, we do need to
+                      // specify "stroke-miterlimit".
+    write_attr_dbl(f, "stroke-miterlimit", lmitre);
```

Comment:
Urgh, that's super annoying. I'd really rather be consistent everywhere, so I think that means we need to change everything to use inline styles.